This is taken from upstream an will be implemented in cmake 3.22.3.
Prevent error related to Qt5 testing.

diff -Nur cmake-3.22.2.old/Tests/RunCMake/CMakeLists.txt cmake-3.22.2/Tests/RunCMake/CMakeLists.txt
--- cmake-3.22.2.old/Tests/RunCMake/CMakeLists.txt	2022-01-25 14:45:06.000000000 +0100
+++ cmake-3.22.2/Tests/RunCMake/CMakeLists.txt	2022-02-17 20:50:59.906685411 +0100
@@ -580,12 +580,16 @@
   add_RunCMake_test(SymlinkTrees)
 endif ()
 
-find_package(Qt4 QUIET)
-find_package(Qt5Core QUIET)
-if (QT4_FOUND AND Qt5Core_FOUND AND NOT Qt5Core_VERSION VERSION_LESS 5.1.0)
+if(CMake_TEST_Qt4)
+  find_package(Qt4 QUIET)
+endif()
+if(CMake_TEST_Qt5)
+  find_package(Qt5Core QUIET)
+endif()
+if (CMake_TEST_Qt4 AND CMake_TEST_Qt5 AND QT4_FOUND AND Qt5Core_FOUND AND NOT Qt5Core_VERSION VERSION_LESS 5.1.0)
   add_RunCMake_test(IncompatibleQt)
 endif()
-if (QT4_FOUND)
+if (CMake_TEST_Qt4 AND QT4_FOUND)
   add_RunCMake_test(ObsoleteQtMacros -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE})
 endif()
 
